src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs

```diff
@@ -0,0 +1,149 @@
+            if (!interfaces.TryGetValue(name, out lni))
+            {
+                lni = new LinuxNetworkInterface(name);
+                interfaces.Add(name, lni);
+            }
+
+            return lni;
+        }
+
+        public override bool SupportsMulticast { get { return GetSupportsMulticast(); } }
+
+        private bool GetSupportsMulticast()
+        {
+            string path = Path.Combine(LinuxNetworkFiles.SysClassNetFolder, _name, "flags");
+            string fileContents = File.ReadAllText(path).Trim();
+            LinuxNetDeviceFlags flags = (LinuxNetDeviceFlags)Convert.ToInt32(fileContents, 16);
+            return (flags & LinuxNetDeviceFlags.IFF_MULTICAST) == LinuxNetDeviceFlags.IFF_MULTICAST;
+        }
+
+        public override IPInterfaceProperties GetIPProperties()
+        {
+            return new LinuxIPInterfaceProperties(this);
+        }
+
+        public override IPInterfaceStatistics GetIPStatistics()
+        {
+            return new LinuxIPInterfaceStatistics(_name);
+        }
+
+        public override OperationalStatus OperationalStatus
+        {
+            get
+            {
+                // /sys/class/net/<name>/operstate
+                string path = Path.Combine(LinuxNetworkFiles.SysClassNetFolder, _name, "operstate");
+                string state = File.ReadAllText(path).Trim();
+                return MapState(state);
+            }
+        }
+
+        // Maps values from /sys/class/net/<interface>/operstate to OperationStatus values.
+        private OperationalStatus MapState(string state)
+        {
+            // TODO: Figure out the possible values that Linux might return.
+            switch (state)
+            {
+                case "up":
+                    return OperationalStatus.Up;
+                case "down":
+                    return OperationalStatus.Down;
+                default:
+                    return OperationalStatus.Unknown;
+            }
+        }
+
+        public override bool Supports(NetworkInterfaceComponent networkInterfaceComponent)
+        {
+            Sockets.AddressFamily family =
+                (networkInterfaceComponent == NetworkInterfaceComponent.IPv4)
+                ? Sockets.AddressFamily.InterNetwork
+                : Sockets.AddressFamily.InterNetworkV6;
+
+            return _addresses.Any(addr => addr.AddressFamily == family);
+        }
+
+        public override string Id { get { throw new PlatformNotSupportedException(); } }
+
+        public override string Description { get { throw new PlatformNotSupportedException(); } }
+
+        public override long Speed
+        {
+            get
+            {
+                try
+                {
+                    string path = Path.Combine(LinuxNetworkFiles.SysClassNetFolder, _name, "speed");
+                    string contents = File.ReadAllText(path);
+                    long val;
+                    if (long.TryParse(contents, out val))
```

Comment:
Nit (style): I usually prefer if (!...) throw. Removes the need of the `else` block and is a bit simpler to separate the error case.